NadekoBot.Core/Modules/Searches/Services/SearchesService.cs

```diff
@@ -360,7 +364,11 @@ public string GetRandomImageUrl(ImageTag tag)
             }
             else if (tag == ImageTag.Goose){
                 return $"https://raw.githubusercontent.com/steggie3/goose-dataset/master/images/goose-mugshot-{_rng.Next(1, max).ToString("0000")}.jpg";
-            }else{
+            }
+            else if (tag == ImageTag.Image){
+                return $"https://picsum.photos/id/{_rng.Next(1, max).ToString()}/2000";
```

Comment:
This is good enough but if you feel like learning a bit more, I think it would be great if you could use the API itself to grab a random image instead of doing the random generation here. The reason why you might want to use their API instead of randomly generating an ID yourself is because they don't make any guarantees (none that I can find) about their IDs not having any gaps in the sequence. If you want to explore this, here are two ways I've come up with.

The first way is to try doing it using https://picsum.photos/seed/picsum/2000 (assuming you want 2000px). The problem you might have run into is that if you paste that link into Discord, it will show one image, but "Show original" takes you to a different one. That's cause the link is simply a redirect and it will give you a new image each time. So if you do it this way, you would have to resolve the redirect and return that, for example using https://stackoverflow.com/questions/704956/getting-the-redirected-url-from-the-original-url.

The other way, which is probably easier, is to use their "static random image endpoint" like https://picsum.photos/seed/RANDOM_STRING_HERE/2000. You still have to generate a random string, but that's straightforward. And even though this link is a redirect, it will always redirect to the same random image. It's easier than the first method cause you don't have to resolve the redirect yourself.

Take a look at these two options and see how it goes.

Reply:
Yeah good idea. I'll have a look at both soon. Thanks for the suggestion

Reply:
Went with the second option